root/static/scripts/edit/URLCleanup.js

```diff
@@ -4690,6 +4690,34 @@ const CLEANUPS: CleanupEntries = {
       return {result: false, target: ERROR_TARGETS.URL};
     },
   },
+  'threads': {
+    match: [new RegExp('^(https?://)?([^/]+\\.)?threads\\.net/', 'i')],
+    restrict: [{...LINK_TYPES.streamingfree, ...LINK_TYPES.socialnetwork}],
+    clean: function (url) {
+      return url.replace(
+        /^(?:https?:\/\/)?(?:www\.)?threads\.net(?:\/#!)?\//,
+        'https://www.threads.net/',
+      );
+    },
+    validate: function (url, id) {
+      const isAProfile = /^https:\/\/www\.threads\.net\/(@[^/]+)$/.test(url);
+      const isAThread = /^https:\/\/www\.threads\.net\/t\/[\w\d]+?$/.test(url);
```

Comment:
1. The quantifier `?` is not wanted here as the thread identifier is mandatory.
2. The expression would not work with thread identifier containing a hyphen `-`.
   Maybe just use the same expression as above?
```suggestion
      const isAThread = /^https:\/\/www\.threads\.net\/t\/[^/]+$/.test(url);
```